libkbfs/unflushed_path_cache.go

```diff
@@ -336,13 +368,50 @@ func (upc *unflushedPathCache) initialize(ctx context.Context,
 		default:
 		}
 
-		log.CDebugf(ctx, "Processing unflushed paths for %d items in "+
-			"the append queue", len(queue))
-		err := addUnflushedPaths(ctx, uid, key, codec, log, queue, cpp,
-			unflushedPaths)
-		if err != nil {
-			return nil, false, err
+		// Do the queued appends up to the first reinitialization.
+		for len(queue) > 0 {
+			var appends []unflushedPathMDInfo
```

Comment:
You're specifically asking why we do the removes after the appends?  (Everything else is FIFO already.) I am trying to amortize the cost of the `cpp.populateChainPaths` call in `addUnflushedPaths()` by doing all the appends together, since that is potentially an expensive call that needs to walk the TLF tree and gather blocks along the way.